dev-tools/smoke_test_rc.py

```diff
@@ -263,6 +265,8 @@ def parse_list(string):
                       help='The Elasticsearch Version to smoke-tests', required=True)
   parser.add_argument('--hash', '-s', dest='hash', default=None, required=True,
                       help='The sha1 short hash of the git commit to smoketest')
+  parser.add_argument('--release-hash', '-r', dest='release_hash', default=None, required=True,
```

Comment:
What does that check actually help us verify? I don't think smoke testing should care about what particular hash was used to build, it should care about whether the artifacts it is told to download work? I view the release as a blob. That particular check requires us to know the "release hash" as you called it here (other unified release stuff calls the version+hash the build id, note that it is not a git hash of any repo), *and* the ES commit hash, but I think it should be easier to run smoke tests. We can do that check (did we put the hash in the jar we built correctly) in tests in ES directly.